demo/fake_server.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Wanda.Executions.FakeServer do
+  @behaviour Wanda.Executions.ServerBehaviour
+
+  import Wanda.Factory
+
+  require Logger
+
+  alias Wanda.Executions
+
+  alias Wanda.Executions.{
+    Execution,
+    Result
+  }
+
+  alias Wanda.Messaging
+
+  @impl true
+  def start_execution(execution_id, group_id, targets, env, config \\ []) do
+    create_fake_execution(execution_id, group_id, targets)
+    Process.sleep(2_000)
+    complete_fake_execution(execution_id, group_id, targets, :passing)
+
+    :ok
+  end
+
+  @impl true
+  def receive_facts(execution_id, group_id, agent_id, facts) do
+    :ok
+  end
+
+  defp create_fake_execution(execution_id, group_id, targets) do
+    insert(:execution,
+      status: :running,
+      execution_id: execution_id,
+      group_id: group_id,
+      targets: Enum.map(targets, &Map.from_struct/1)
```

Comment:
We need to be careful with this 1:1 targets conversion.
We could fall in previous errors, like sending back check results that are not even in the catalog. You might want to visit how it works in the real execution.

Reply:
Is this relevant for an environment in which we control what gets send? I don't think this code will be ever used outside of demo / development purposes

Reply:
well, we already saw errors here @rtorrero 
it is not about using the code somewhere else (which won't happen), it is about having a potential broken view if we receive some check id that it is not in the catalog.
Anyway, I don't want to extend this discussion. I see the risk. If you are fine with the solutions, so be it